### ISSUE TITLE
@nodoc on overridden methods only half-works

### ISSUE THREAD

**Issue by tvolkert** [dart-lang/dartdoc#1337]
For example:

```
abstract class Matcher {
  /// This does the matching of the actual vs expected values.
  /// [item] is the actual value. [matchState] can be supplied
  /// and may be used to add details about the mismatch that are too
  /// costly to determine in [describeMismatch].
  bool matches(item, Map matchState);

  /// This builds a textual description of the matcher.
  Description describe(Description description);
}

abstract class NoDescriptionMatcher extends Matcher {
  /// @nodoc
  @override
  Description describe(Description description) => description;
}

class MyMatcher extends NoDescriptionMatcher {
}
```

If you generate dartdoc for this class structure, the HTML page for `MyMatcher` correctly does not show the dartdoc for the `describe` method.  However, the HTML page for `NoDescriptionMatcher` *does*

**Comment by jcollins-g** [comment]
`@nodoc`'s current implementation is too simple, and it doesn't take either inheritance nor dartdoc's ability to import docs from parent classes into account.

**Comment by jcollins-g** [comment]
I believe this has been fixed since v0.14.1, but didn't get closed for some reason.
